### ISSUE TITLE
Error when installing the polyfill on an iframe

### ISSUE THREAD

**Issue by gkubisa** [WebReflection/document-register-element#149]
```html
<iframe id="frame"></iframe>
```

```javascript
import installCustomElements from 'document-register-element/pony'
const iframeWindow = window.frames['frame'].contentWindow || window.frames['frame'].window
installCustomElements(iframeWindow)
```

`Uncaught TypeError: Illegal constructor` is thrown by [`new DRE()`](https://github.com/WebReflection/document-register-element/blob/d0456740ea4bf321b55ecf550a95179e8de8a3bf/pony/index.js#L1475).

Reproduced in Chrome 68, Ubuntu 16.04.

The error no longer happens, when I replace `HTMLAnchorElement` with `window.HTMLAnchorElement` [here](https://github.com/WebReflection/document-register-element/blob/d0456740ea4bf321b55ecf550a95179e8de8a3bf/pony/index.js#L1470) and [here](https://github.com/WebReflection/document-register-element/blob/d0456740ea4bf321b55ecf550a95179e8de8a3bf/pony/index.js#L1481).

**Comment by WebReflection** [comment]
You need patched constructors in the new iframe too, testing what's been patched already wouldn't help much making it work in there, I guess ... or would it?

**Comment by gkubisa** [comment]
At the time [`new DRE()`](https://github.com/WebReflection/document-register-element/blob/d0456740ea4bf321b55ecf550a95179e8de8a3bf/pony/index.js#L1475) is called, nothing has been polyfilled yet - we're just checking, if the polyfill is necessary in a browser which supports autonomous custom elements but might not support built-in custom elements.

There are separate `HTMLAnchorElement` constructors in the main window and in the iframe. If I execute `installCustomElements` in the context of the main window, then `HTMLAnchorElement` is going to always be the constructor from the main window, regardless of what object I pass in as the [`window` parameter](https://github.com/WebReflection/document-register-element/blob/d0456740ea4bf321b55ecf550a95179e8de8a3bf/pony/index.js#L23). However, if I'm polyfilling an iframe's window, the iframe's `HTMLAnchorElement` should be used to avoid the "Illegal constructor" error. Inside `installCustomElements`, `window.HTMLAnchorElement` is the right constructor to use.

In other words, we register a test custom element on the iframe [here](https://github.com/WebReflection/document-register-element/blob/d0456740ea4bf321b55ecf550a95179e8de8a3bf/pony/index.js#L1472), so we should use a constructor defined by that iframe too, which is `window.HTMLAnchorElement`. `HTMLAnchorElement !== window.HTMLAnchorElement` inside `installCustomElements(window, polyfill)`, if the `window` param is anything else than the global object (the main window).

**Comment by WebReflection** [comment]
it should work in `v1.11.1`
